state/core/src/main/java/com/expediagroup/streamplatform/streamregistry/state/EntityViewUpdater.java

```diff
@@ -51,45 +55,62 @@
     }
   }
 
+  <K extends Entity.Key<S>, S extends Specification> Optional<Entity<K, S>> purge(K key) {
+    val stateEntity = Optional.ofNullable(entities.get(key))
+      .filter(it -> it.deleted);
+    stateEntity.ifPresent(it -> {
+      entities.remove(key);
+      log.debug("Purged entity for {}", key);
```

Comment:
This is minor, but it's more Splunk friendly if you do something like 
```suggestion
      log.debug("Purged entity for key={}", key);
```

Reply:
Although I did notice that we don't tend to log key value pairs :'(

Reply:
done